library/core/src/array/mod.rs

```diff
@@ -364,3 +364,61 @@ macro_rules! array_impl_default {
 }
 
 array_impl_default! {32, T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T T}
+
+#[cfg(not(bootstrap))]
+#[lang = "array"]
+impl<T, const N: usize> [T; N] {
+    /// Returns an array of the same size as `self`, with function `f` applied to each element.
+    /// The closure will be called on elements 0 up to but excluding N.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #![feature(array_map)]
+    /// let x = [1, 2, 3];
+    /// let y = x.map(|v| v + 1);
+    /// assert_eq!(y, [2, 3, 4]);
+    ///
+    /// let x = [1, 2, 3];
+    /// let mut temp = 0;
+    /// let y = x.map(|v| { temp += 1; v * temp });
+    /// assert_eq!(y, [1, 4, 9]);
```

Comment:
Might I suggest an example where we change the type of the array? Maybe, something like `["Ferris", "is", "cute"].map(|s| s.len())` -> `[6, 2, 4]`.